src/gpu/nvidia/cudnn_inner_product.cpp

```diff
@@ -156,15 +156,65 @@ status_t cudnn_inner_product_bwd_weights_t::execute(
 
         if (wei_sz != 0) {
             auto status = cuda_stream->interop_task([&](::sycl::handler &cgh) {
-                auto diff_wei_acc = CTX_OUT_ACCESSOR(DNNL_ARG_DIFF_WEIGHTS);
-                cgh.fill(diff_wei_acc, static_cast<uint8_t>(0));
+                auto *mem_diff_wei
```

Comment:
`sycl_stream_t` has a [fill](https://github.com/oneapi-src/oneDNN/blob/002a6697c55ffa69bdcded42f467175dcd865f9b/src/sycl/sycl_stream.hpp#L205) function. I believe it can be used to zero a memory storage. It has one nuance, it creates a dependency on the previous event. For example, if it's called for weights and bias here then the calls will be serialized.
```cpp
auto status = cuda_stream->fill(CTX_OUT_STORAGE(DNNL_ARG_DIFF_WEIGHTS), 0, wei_sz);
```

If you want to have more control over dependencies then you can do something like this:
```cpp
auto arg_diff_wei = CTX_OUT_SYCL_MEMORY(DNNL_ARG_DIFF_WEIGHTS);
// `is_usm()` should be introduced to `sycl_memory_arg_t`.
if (arg_diff_wei.is_usm()) {
    // `usm()` should be introduced to `sycl_memory_arg_t`.
    cgh.fill(arg_diff_wei.usm(), 0, wei_sz);
} else {
    // accessor()` should be introduced to `sycl_memory_arg_t`.
    cgh.fill(arg_diff_wei.accessor(), 0);
}
```

Reply:
Done. That is much simpler. Thanks.